lua/lspconfig/configs.lua

```diff
@@ -257,16 +257,23 @@ function configs.__newindex(t, config_name, config_def)
       end
       local buf_path = util.path.sanitize(bufname)
 
-      if get_root_dir then
-        root_dir = get_root_dir(buf_path, bufnr)
-      end
+      coroutine.resume(coroutine.create(function()
+        local root_dir
+        if get_root_dir then
+          root_dir = get_root_dir(buf_path, bufnr)
+        end
 
-      if root_dir then
-        manager.add(root_dir, false, bufnr)
-      elseif config.single_file_support then
-        local pseudo_root = #bufname == 0 and uv.cwd() or util.path.dirname(buf_path)
-        manager.add(pseudo_root, true, bufnr)
-      end
+        if root_dir then
+          vim.schedule(function()
+            manager.add(root_dir, false, bufnr)
+          end)
```

Comment:
Avoid `vim.schedule` when it's not necessary? Same on the branch below
```suggestion
          if not vim.in_fast_event() then
            manager.add(root_dir, false, bufnr)
          else
            vim.schedule(function()
              manager.add(root_dir, false, bufnr)
            end)
          end
```
Or it could be even something like this, maybe even a new utility function that does this:
```lua
local function cb()
  manager.add(root_dir, false, bufnr)
end
if vim.in_fast_event() then
  vim.schedule(cb)
else
  cb()
end
```

Reply:
make sense